src/Compilers/VisualBasic/Portable/Utilities/TypeUnification.vb

```diff
@@ -107,6 +107,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                     Dim nt1 As NamedTypeSymbol = DirectCast(t1.Type, NamedTypeSymbol)
                     Dim nt2 As NamedTypeSymbol = DirectCast(t2.Type, NamedTypeSymbol)
+
+                    If nt1.IsTupleType Then
+                        If Not nt2.IsTupleType Then
+                            Return False
+                        End If
```

Comment:
I think the code would be more robust if we simply dig through tuples GetTupleUnderlyingTypeOrSelf if either nt1, or nt2 is a tuple.